app/src/main/java/ch/sdp/vibester/fragment/ScoreBoardFragment.kt

```diff
@@ -26,42 +28,46 @@ import com.google.firebase.database.ValueEventListener
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
-class ScoreBoardActivity : AppCompatActivity(), OnItemClickListener {
+class ScoreBoardFragment : Fragment(), OnItemClickListener, View.OnClickListener{
     private val dbRef: DatabaseReference = Database.get().getReference("users")
     private var players: MutableList<User>? = null
     private var userScoreboardAdapter: UserScoreboardAdapter? = null
     private var genre: String = ""
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        setContentView(R.layout.scoreboard_scoreboard)
-        setSupportActionBar(findViewById(R.id.toolbar))
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_scoreboard, container, false)
+        view.findViewById<Button>(R.id.kpopButton).setOnClickListener(this)
+        view.findViewById<Button>(R.id.rockButton).setOnClickListener(this)
+        view.findViewById<Button>(R.id.btsButton).setOnClickListener(this)
+        view.findViewById<Button>(R.id.topTracksButton).setOnClickListener(this)
+        view.findViewById<Button>(R.id.imagDragonsButton).setOnClickListener(this)
+        view.findViewById<Button>(R.id.billieEilishButton).setOnClickListener(this)
+        return view
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
         setupRecycleView()
         players = ArrayList()
     }
 
-    fun selectScoreboard(view: View) {
-        var sortedBy = "scores/"
-        // can't use R.string for genre (getOrDefault in setScore in UserScoreboardAdapter)
-        when (view.id) {
-            R.id.btsButton -> {sortedBy += R.string.bts; genre = "BTS"}
-            R.id.kpopButton -> {sortedBy += R.string.kpop; genre = "kpop"}
-            R.id.imagDragonsButton -> {sortedBy += R.string.imagine_dragons; genre = "Imagine Dragons"}
-            R.id.billieEilishButton -> {sortedBy += R.string.billie_eilish; genre = "Billie Eilish"}
-            R.id.rockButton -> {sortedBy += R.string.rock; genre = "rock"}
-            R.id.topTracksButton -> {sortedBy += R.string.top_tracks; genre = "top tracks"}
-        }
+    fun selectScoreboard() {
```

Comment:
Why keeping this function public ? 
Also if we keep it public, maybe considering adding one line of javadoc to describe its behavior

Reply:
it should be private, fixed!